### PR TITLE
Add function_call to FakeListChatModel

### PR DESCRIPTION
**Pull request by NarekA**
This PR allows users to simulate function calls using the `FakeListChatModel` by passing in dictionary values as the responses.

### CODE REVIEW

File: libs/langchain/langchain/chat_models/fake.py [hunk 1]
```diff
@@ -53,7 +53,17 @@ def _stream(
         for c in response:
             if self.sleep is not None:
                 time.sleep(self.sleep)
-            yield ChatGenerationChunk(message=AIMessageChunk(content=c))
+            if isinstance(response, dict):
```
Comment (baskaryan) [inline libs/langchain/langchain/chat_models/fake.py:56]: should this be checking if response is a dict or if c is a dict? if the former, does it need to be in the for loop?

Comment (NarekA) [reply libs/langchain/langchain/chat_models/fake.py]: Yes, good catch! This really needs to be in the `_generate` method. Can you take a look at the latest version? I will add tests soon.

**Comment by leo-gan** [comment]
@NarekA Hi , could you, please, resolve the merging issues and address the last comments (if needed)? After that, ping me and I push this PR for the review. Thanks!

**Comment by efriis** [comment]
Looks like this isn't being worked on anymore. I'll close for now, and if you resume work, feel free to reopen!

**Pull request closed without merging**